src/ray/common/task/task.h

```diff
@@ -74,6 +113,9 @@ class Task {
   /// Task execution specification, consisting of all dynamic/mutable
   /// information about this task determined at execution time.
   TaskExecutionSpecification task_execution_spec_;
+  /// If this is a vector task, list of all task specs including the first one.
+  /// In this case task_spec_ == task_spec_vector_[0];
+  std::shared_ptr<std::vector<TaskSpecification>> task_spec_vector_;
```

Comment:
Can probably make this a uniqu_ptr?

Reply:
Problem is task must be copyable. Arguably we should have been passing around unique_ptr<Task> instead of Task, but this would be quite a huge refactoring to do.

Reply:
Ok, then let's leave the `shared_ptr`.